packages/mui-material/src/styles/createTheme.test.js

```diff
@@ -5,10 +5,43 @@ import Button from '@mui/material/Button';
 import Box from '@mui/material/Box';
 import { ThemeProvider, createTheme, styled } from '@mui/material/styles';
 import { deepOrange, green } from '@mui/material/colors';
+import createPalette from './createPalette';
+
+const lightPalette = createPalette({ mode: 'light' });
+const darkPalette = createPalette({ mode: 'dark' });
 
 describe('createTheme', () => {
   const { render } = createRenderer();
 
+  it('should not have custom properties', () => {
+    const theme = createTheme({ cssVariables: false });
+    expect(theme.cssVariables).to.equal(false);
+    expect('vars' in theme).to.equal(false);
+  });
+
+  it('color schemes dark: true', () => {
+    const theme = createTheme({ cssVariables: false, colorSchemes: { dark: true } });
+    const { light, dark } = theme.colorSchemes || {};
+    expect(light?.palette.primary.main).to.deep.equal(lightPalette.primary.main);
+    expect(dark?.palette.primary.main).to.deep.equal(darkPalette.primary.main);
+  });
+
+  it('color schemes light: true', () => {
+    const theme = createTheme({
+      cssVariables: false,
+      colorSchemes: { light: true },
+      palette: { mode: 'dark' },
+    });
```

Comment:
What should this generate, I am a bit confused. Is this even a valid input?

Reply:
This case test a default dark app with light mode.

Reply:
It's similar to:

```js
createTheme({
  cssVariables: false,
  colorSchemes: { dark: true },
  palette: { mode: 'light' },
});
```

Reply:
> What should this generate, I am a bit confused. Is this even a valid input?

Yes, you can think of having the provided `palette` as the default color scheme.

Reply:
What I understand:

The `palette` values will continue working the same way as before, but the values will also be copied into `colorSchemes`

Is that correct?

If it's correct, what's the use of copying the palette values into the color schemes? In other words, how is `theme.colorSchemes` going to be used in our codebase or in users codebases when `cssVariables: false`?

Is the idea to use `colorSchemes` from now on and eventually stop using palette?

Reply:
> The palette values will continue working the same way as before, but the values will also be copied into colorSchemes Is that correct?

Yes, that's my purpose. My intention is the `colorSchemes` is a new feature to create a second and more color schemes. `palette` is considered a default color scheme and will continue to work at least in v6.

> If it's correct, what's the use of copying the palette values into the color schemes? In other words, how is theme.colorSchemes going to be used in our codebase or in users codebases when cssVariables: false?

Got it. The `ThemeProvider` contains a state that store a user selected mode, a part of the theme is calculated from `theme.colorSchemes[selectedMode]`. For example, when user is in dark mode, the theme that passed through context will be:

```js
{
  …theme.colorSchemes.dark,
  typography: …,
  spacing: …,
  breakpoints: …,
}
```

Reply:
This makes sense to me 👌🏼 

When we discuss the v7 theme structure, we can discuss the plan for `theme.palette`.